core/src/main/java/io/grpc/ServiceDescriptor.java

```diff
@@ -108,4 +132,89 @@ static void validateMethodNames(String serviceName, Collection<MethodDescriptor<
           "duplicate name %s", method.getFullMethodName());
     }
   }
+
+  /**
+   * Creates a new builder for a {@link ServiceDescriptor}.
+   *
+   * @since 1.1.0
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2666")
+  public static Builder newBuilder() {
```

Comment:
Comparing to protobuf doesn't help the discussion. It doesn't matter what protobuf does because it could not have done what I'm describing. If I was making the protobuf builder I would have done exactly like they would have, because there is no other option.

More examples requiring:
 * [ProcessBuilder](http://docs.oracle.com/javase/8/docs/api/java/lang/ProcessBuilder.html)
 * [DateTimeFormatterBuilder](http://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatterBuilder.html)'s `toFormatter(Locale)`

I (surprisingly, honestly) have not found an example in a similar circumstance that follows your recommendation.